### PR TITLE
2006 trigger error on url in xform title

### LINKED ISSUE #2006: Trigger error when xform title matches a url
### Environmental Information
All
### Onadata version:
master 

### Problem description
Prevent HTML injection

### Expected behavior
Trigger an error when xform title matches url

### Steps to reproduce the behavior
Use PUT request to update form

### Additional Information

_Logs, [related issues](github.com/onaio/onadata/issues), weird / out of place occurrences, local settings, possible approach to solving this..._

### PR DESCRIPTION
**Pull request by ivermac**
### Changes / Features implemented
Trigger error when xform title matches a url

### Steps taken to verify this change does what is intended
Added a test to check that an error is triggered when xform title matches a url

### Side effects of implementing this change

Closes #2006

### CODE REVIEW

File: onadata/apps/logger/models/xform.py [hunk 1]
```diff
@@ -818,6 +818,13 @@ def _set_title(self):
                 _("Title shouldn't have any invalid xml "
                   "characters ('>' '&' '<')"))
 
+        pattern = str("http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|"
```
Comment (DavisRayM) [inline onadata/apps/logger/models/xform.py:821]: Should we update the pattern to also match URLs that don't include the HTTP/HTTPS protocols?

Screenshot 2021-02-01 at 12 10 31 PM

**Review by DavisRayM — APPROVED** [review]
LGTM, just a few suggestions

File: onadata/apps/logger/models/xform.py (hunk 1 above)
Comment (ivermac) [reply onadata/apps/logger/models/xform.py:821]: Yes we should. Let me update

**Review by DavisRayM — APPROVED** [review]
LGTM

**Pull request merged**